core/trino-main/src/test/java/io/trino/util/TestDateTimeUtils.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.util;
+
+import org.testng.annotations.Test;
+
+import java.time.DateTimeException;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+
+public class TestDateTimeUtils
+{
+    @Test
+    public void testTryParseDateYyyyMmDd()
+    {
+        // valid dates
+        Optional<Integer> days = DateTimeUtils.tryParseDateYyyyMmDd("1970-01-01");
+        assertEquals(0, days.get().intValue());
+        days = DateTimeUtils.tryParseDateYyyyMmDd("1970-02-01");
+        assertEquals(31, days.get().intValue());
+        days = DateTimeUtils.tryParseDateYyyyMmDd("1969-12-01");
+        assertEquals(-31, days.get().intValue());
+        days = DateTimeUtils.tryParseDateYyyyMmDd("2022-02-28");
+        assertEquals(19051, days.get().intValue());
+        days = DateTimeUtils.tryParseDateYyyyMmDd("0000-01-01");
+        assertEquals(-719528, days.get().intValue());
+        days = DateTimeUtils.tryParseDateYyyyMmDd("9999-12-31");
+        assertEquals(2932896, days.get().intValue());
+
+        // format invalid
+        days = DateTimeUtils.tryParseDateYyyyMmDd("1970-2-01");
```

Comment:
Add an invalid case for every `if` statement in the `tryParseDateYyyyMmDd` e.g. `197A-01-01`, `197#-01-01` so that all code paths are covered.

Reply:
a test case added for each code path in 921b9c23e9894fdc57a581a29cb5d6aee89a99f9